src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/NomisEventOutcomeMapper.java

```diff
@@ -17,48 +16,45 @@
 }
 
 class NomisEventOutcomeMapper {
-    private static final HashMap<Long, EventOutcome> codeMap = new HashMap<>();
+    private static final HashMap<AbsentReason, EventOutcome> eventOutcomes = new HashMap<>();
 
     NomisEventOutcomeMapper() {
-        addPaidAttendanceMapping("ATT", "STANDARD");
+        eventOutcomes.put(AbsentReason.AcceptableAbsence, new EventOutcome("ACCAB", null));
+        eventOutcomes.put(AbsentReason.NotRequired, new EventOutcome("NREQ", null));
 
-        addPaidAbsenceMapping(AbsentReason.AcceptableAbsence,"ACCAB");
-        addPaidAbsenceMapping(AbsentReason.NotRequired,"NREQ");
+        eventOutcomes.put(AbsentReason.SessionCancelled, new EventOutcome("CANC", null));
+        eventOutcomes.put(AbsentReason.RestInCell, new EventOutcome("REST", null));
 
-        addUnPaidAbsenceMapping(AbsentReason.SessionCancelled, "CANC");
-        addUnPaidAbsenceMapping(AbsentReason.RestInCell,  "REST");
-        addUnPaidAbsenceMapping(AbsentReason.Sick, "REST");
-        addUnPaidAbsenceMapping(AbsentReason.RestDay, "REST");
+        eventOutcomes.put(AbsentReason.Sick, new EventOutcome("REST", null));
+        eventOutcomes.put(AbsentReason.RestDay, new EventOutcome("REST", null));
 
-        addUnPaidAbsenceMapping(AbsentReason.Refused,  "UNACAB");
-        addUnPaidAbsenceMapping(AbsentReason.UnacceptableAbsence,  "UNACAB");
-    }
+        eventOutcomes.put(AbsentReason.Refused, new EventOutcome("UNACAB", null));
+        eventOutcomes.put(AbsentReason.UnacceptableAbsence, new EventOutcome("UNACAB", null));
 
-    private void addPaidAttendanceMapping(final String eventOutcome, final String performance) {
-        codeMap.put(getKey(null, true, true), new EventOutcome(eventOutcome, performance));
     }
 
-    private void addUnPaidAbsenceMapping(final AbsentReason reason, final String eventOutCome) {
-        codeMap.put(getKey(reason, false, false), new EventOutcome(eventOutCome, null));
-    }
+    EventOutcome getEventOutcome(final AbsentReason reason, final boolean attended, final boolean paid) {
```

Comment:
I'm planning on catching these exceptions in the controller, mapping them into 400 bad requests.